codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -952,6 +952,16 @@ private T thisT() {
         return (T) this;
     }
 
+    /**
+     * Returns a deep copy of this instance.
+     */
+    public DefaultHeaders<K, V, T> copy() {
+        DefaultHeaders<K, V, T> copy = new DefaultHeaders<K, V, T>(
+                hashingStrategy, valueConverter, nameValidator, entries.length);
+        copy.set(this);
```

Comment:
`set` will also do `clear` which will do some unnecessary work (array filling, pointer updating, assignments). We should be able to just use `addImpl` instead of `set`.